janggi/gameboard.py

```diff
@@ -319,17 +324,10 @@ def hypothetical_move(self, start, end):
         self.set_square_contents(end, piece_obj)
         piece_obj.set_position(end)
 
-        # get color from starting piece
-        check_color = None
-        if piece_obj.get_color() == "r":
-            check_color = "red"
-        elif piece_obj.get_color() == "b":
-            check_color = "blue"
-
         # run is_in_check on the current player,
         # if in check, set valid_move to FALSE
         # else set valid_move to TRUE
-        if self.is_in_check(check_color):
+        if self.is_in_check(piece_obj.get_color()):
```

Comment:
Why can't we just use `color` directly?

```suggestion
        if self.is_in_check(color):
```

Reply:
I think because ``hypothetical_move`` doesn't have a ``color`` parameter

Reply:
Oh, it's a different method than the previous diff. Sorry, still getting used to the GitHub interface.